presto-main/src/main/java/io/prestosql/execution/buffer/ArbitraryOutputBuffer.java

```diff
@@ -474,6 +481,67 @@ public String toString()
         }
     }
 
+    public enum ClientBuffersReorderStrategy {
```

Comment:
I'm not sure we really need different reordering strategies instead of simple shuffle. It seems that simple reorder should be sufficient as workers should back-pressure when they are processing data slowly.

Reply:
Yes. I think this pull request is more like an issue, and help community to find a final solution.